Tests/NIOTests/BootstrapTest.swift

```diff
@@ -196,7 +197,7 @@ class BootstrapTest: XCTestCase {
                 }
                 return self.freshEventLoop().makeSucceededFuture(())
             }
-            .withBoundSocket(descriptor: socket)
+            .withBoundSocket(socket: socket)
```

Comment:
we need new tests that test _all_ deprecated functionality. It's super risky for us to merge this because no code is exercising those. 

The tests that test deprecated functionality should look like

```
    @available(*, deprecated, message: "deprecated because it tests deprecated functionality")
    func testDeprecatedXYZ() {
```